app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt

```diff
@@ -330,6 +340,14 @@ class AddLoginFragment : Fragment(R.layout.fragment_add_login), MenuProvider {
         }
     }
 
+    private fun hostnameEnsureProtocol(hostname: String): String {
```

Comment:
Could we use a focus listener to set the `https` protocol immediately after the user finishes editing the site url - similar to how it functions on Desktop?
Just to have the protocol clear for the user.
I imagine a case in which the user might omit the protocol, we would add "https" without immediately showing it but the website should user "http". If we show this immediately the user will have the chance to update "https" to "http" immediately.

Reply:
I tried that, but there's an issue with Android here in that tapping on the save button immediately after editing the hostname results in the focus listener being called _after_ the save button action is performed -- in this case if the string does not already contain a valid protocol then the backend silently drops the entire login!

I agree though that giving the user a chance to edit is useful. 2 options here:

1) Force the user to tap save again after auto-adding `https` -- I think this inconsistency would be annoying for users.
2) Use both a focus listener and the current PR's `hostnameEnsureProtocol` method -- this would give users that enter/edit the hostname first a chance to see the `https` addition as well as ensure that no invalid hostnames get silently dropped.

Something else to note is that desktop Firefox displays the login detail page right after adding it while Fenix just pops you back into the saved logins list. I think the auto-addition would be smoother if the user would be able to instantly see the new login after creating it. That is, if they have `example.com` and tap the save button, they would be presented with the `LoginDetailFragment` that includes `Site: https://example.com`, which they can then change if they wish.

Reply:
> Something else to note is that desktop Firefox displays the login detail page right after adding it while Fenix just pops you back into the saved logins list. I think the auto-addition would be smoother if the user would be able to instantly see the new login after creating it. That is, if they have example.com and tap the save button, they would be presented with the LoginDetailFragment that includes Site: https://example.com, which they can then change if they wish. 

Seems like something that may worth a new ticket and discussion with UX.

Reply:
> Could we extract the host verification in a separate method which would be executed also when the save button is tapped such that we'd execute the save action only if the host verification passes?

Yes, though with the current solution of focus listener + `hostnameEnsureProtocol` the save button should only be enabled when the login is either valid by itself, or valid with the addition of `https://` (which `hostnameEnsureProtocol` takes care of).

Reply:
Thank you! Tested the version with the new focus listener and it really improves the entire experience.
Asked UX to confirm the scenario of automatically adding the `https` protocol when the save button is pressed (mentioned that it's the same behavior on desktop).